pysqldb3/geopackage.py

```diff
@@ -201,69 +199,62 @@ def shp_to_gpkg(self, shp_name, gpkg_tbl = None, print_cmd=False):
-                                                gpkg_tbl = gpkg_tbl,
-                                                export_path=self.path)
-
-        if print_cmd:
-            print(print_cmd_string([self.dbo.password], self.cmd))
+                                                gpkg_tbl = gpkg_tbl)
 
         try:
-            ogr_response = subprocess.check_output(shlex.split(self.cmd.replace('\n', ' ')), stderr=subprocess.STDOUT)
+            ogr_response = subprocess.check_output(shlex.split(cmd.replace('\n', ' ')), stderr=subprocess.STDOUT)
             print(ogr_response)
         except subprocess.CalledProcessError as e:
             print("Ogr2ogr Output:\n", e.output)
             print('Ogr2ogr command failed. The Geopackage/feature class was not written.')
-            raise subprocess.CalledProcessError(cmd=print_cmd_string([self.dbo.password], self.cmd), returncode=1)
-
         return
 
-    def gpkg_to_shp(self, print_cmd=False):
-        
+    def gpkg_to_shp(self, gpkg_tbl = None, export_path = None, cmd = None):
         """
         Converts a Geopackage to a Shapefile.
         The output Shapefile name will match the name of the geopackage table to be copied.
 
-        :param gpkg_name: file name for geopackage input (should end in .gpkg)
-
-        :param export_path: The folder directory to place the shapefiles output.
+        :param gpkg_name (str): file name for geopackage input (should end in .gpkg)
+        :param gpkg_tbl (str): OPTIONAL specific table within the geopackage to convert to a Shapefile
+        :param export_path: str The folder directory to place the shapefiles output.
                             You cannot specify the shapefiles' names as they are copied from the table names within the geopackage.
-
-        :param gpkg_tbl: OPTIONAL. Name of the single table in the geopackage that you want to convert to a Shapefile.
-                         Do not add gpkg_tbl argument if you want to copy all tables within the geopackage input.
-
-        :param print_cmd: Print command line query
+        :param print_cmd (bool): Print command line query
         """
         
         assert self.gpkg_name[-5:] == '.gpkg', "The input file should end with .gpkg . Please check your input."
 
         self.path, gpkg = parse_gpkg_path(self.path, self.gpkg_name)
 
+        # set a default self.path for the gpkg input if it exists
         if not self.path:
             self.path = file_loc('folder')
 
-        full_path = os.path.join(self.path, self.gpkg_name)
+        # set an export path to the gpkg if it is not manually set up
+        if not export_path:
+            export_path = self.path
 
         ### bulk upload capabilities ### 
         
-        if not self.gpkg_tbl:
+        if not gpkg_tbl:
 
             #######
             # if no table name listed in argument, then we assume bulk upload
             # this prompts us to count the number of tables within the geopackage to confirm bulk upload
```

Comment:
why do you need to count the tables?

Reply:
Will delete this. Had the count in an earlier iteration of the `.py` file